### PR TITLE
feature: Added C++ solution for USACO-380

### PR DESCRIPTION
**Pull request by naveen-khuntey**
- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/silver/usaco-380.mdx [hunk 1]
```diff
@@ -123,4 +123,84 @@ public class CrossCountrySkiing {
 
 </JavaSection>
 
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+#define MAX 501
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-380.mdx:131]: is this +1 necessary

File: solutions/silver/usaco-380.mdx [hunk 2]
```diff
@@ -123,4 +123,84 @@ public class CrossCountrySkiing {
...
+		pair<int, int> p = q.front();
+		q.pop();
+
+		for (int i = 0; i < 4; i++) {
+			int ny = p.first + dy[i];
+			int nx = p.second + dx[i];
+			if (ny >= 0 && ny < m && nx >= 0 && nx < n)
+				// if the target cell is not visited before
+				//    and the elevation difference is within D
+				//    push the cell into the queue
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-380.mdx:156]:
```cpp
/*
 * use multiline comments
 * for thngs like these
 */
```

File: solutions/silver/usaco-380.mdx [hunk 3]
```diff
@@ -123,4 +123,84 @@ public class CrossCountrySkiing {
...
+			if (wp[i][j] && !mark[i][j]) return false;
+	return true;
+}
+
+int main() {
+	freopen("ccski.in", "r", stdin);
+	freopen("ccski.out", "w", stdout);
+	cin >> m >> n;
+	for (int i = 0; i < m; i++)
+		for (int j = 0; j < n; j++) cin >> grid[i][j];
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-380.mdx:185]: braces

File: solutions/silver/usaco-380.mdx [hunk 4]
```diff
@@ -123,4 +123,84 @@ public class CrossCountrySkiing {
...
+
+	// binary search
+	int l = 0, r = 1e9;
+	while (l < r) {
+		int d = (l + r) / 2;
+		if (reachable(d)) r = d;
+		else l = d + 1;
+	}
+
+	cout << l << "\n";
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-380.mdx:201]:
```suggestions
	cout << l << endl;
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
you can do all this or just have me make the changes for you

**Comment by naveen-khuntey** [comment]
I have applied all the changes.

**Comment by naveen-khuntey** [comment]
I have applied all the changes.

**Comment by SansPapyrus683** [comment]
i feel like this can go a lot faster if you just allow me to format your code
so can i?

**Comment by naveen-khuntey** [comment]
Yes please go ahead

**Review by SansPapyrus683 — DISMISSED** [review]

**Review by ryanchou-dev — APPROVED** [review]
looks great! thanks for the contribution :yayy:

**Pull request merged**